modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/sidebar/comments/FragmentComments.es.js

```diff
@@ -71,6 +71,7 @@ const FragmentComments = props => {
 					fragmentEntryLinkId={fragmentEntryLink.fragmentEntryLinkId}
 				/>
 
+				{/* Intentar optimizar esto (useMemo??) */}
 				{[...fragmentEntryLinkComments].reverse().map(comment => (
```

Comment:
Reverse inside the map 🤔 
```js
{
	fragmentEntryLinkComments.map((_comment, index, arr) => {
		const commentReversed = arr[arr.length - 1 - index];
		return (
			<FragmentComment
				comment={commentReversed}
				fragmentEntryLinkId={fragmentEntryLink.fragmentEntryLinkId}
				key={comment.commentId}
				onDelete={deleteComment}
				onEdit={editComment}
				onEditReply={editCommentReply}
			/>
		);
	});
}
```

Reply:
@p2kmgcl @victorg1991 @wincent

What do you think?

Reply:
This looks ok to me. I don't think there is any "perfect" fix for this on the front-end; they all have different strengths of weaknesses:

- This approach:
  - Pro: small, minimal.
  - Con: "clever".

vs:

- Extracting this out into another function or component:
  - Pro: less "clever".
  - Con: more verbose; have to pass (lots of?) props to it.

```javascript
function FragmentCommentsList(comments, ...) {
  const comments = [];

  for (let i = comments.length - 1; i >= 0; i--) {
    comments.push(<Fragment .../>);
  } 

  return comments;
}
```

- Hosting this up higher in the function body:
  - Pro: less "clever".
  - Con: render logic doesn't read linearly any more.

```javascript
const comments = [];

for (let ...) {
}

return (
    <>
       ...
       {comments}
    </>
  );
```

vs

- Reversing the list and memoizing the result...

vs

- Inlining the `for`-style logic in-place.
- etc, etc...

Reply:
@boton I like that solution, that way we do not need to create extra arrays (which was the main issue, I think). Maybe we could create a technical task to allow comment sorting in backend